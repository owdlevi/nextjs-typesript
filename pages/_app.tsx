// import App from "next/app";
import type { AppProps /*, AppContext */ } from 'next/app'
import Reset from '../components/reset'

function MyApp({ Component, pageProps }: AppProps) {
  return (
    <>
      <Reset />
      <Component {...pageProps} />
    </>
  )
}

export default MyApp
