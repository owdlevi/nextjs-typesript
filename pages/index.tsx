import Head from 'next/head'
import Header from '../components/Header'

export const Home = (): JSX.Element => (
  <>
    <Head>
      <title>Create Next App</title>
      <link rel="icon" href="/favicon.ico" />
    </Head>
    <Header></Header>
    <main>
      <h1>NextJs with typescript</h1>
      <h1>NextJs with typescript</h1>
      <h1>NextJs with typescript</h1>
      <h1>NextJs with typescript</h1>
    </main>
    <footer></footer>
  </>
)

export default Home
