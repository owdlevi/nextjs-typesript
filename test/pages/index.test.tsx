import React from 'react'
import { render, fireEvent } from '../testUtils'
import { Home } from '../../pages/index'
import { getByRole } from '@testing-library/react'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('page has a H1 element', () => {
    const { getByText } = render(<Home />, {})
  })
})
