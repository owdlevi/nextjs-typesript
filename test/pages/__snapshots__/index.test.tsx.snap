// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="jsx-2631374831 container"
  >
    <main
      class="jsx-2631374831"
    />
    <footer
      class="css-d3pce8"
      font-size="24"
    >
      <a
        class="jsx-2631374831"
        href="https://vercel.com?utm_source=create-next-app&utm_medium=default-template&utm_campaign=create-next-app"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by 
        <img
          alt="Vercel Logo"
          class="jsx-2631374831 logo"
          src="/vercel.svg"
        />
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
